codec/proto_codec.go

```diff
@@ -283,7 +275,7 @@ func (g grpcProtoCodec) Marshal(v interface{}) ([]byte, error) {
 	switch m := v.(type) {
 	case proto.Message:
 		return proto.Marshal(m)
-	case ProtoMarshaler:
+	case gogoproto.Message:
 		return g.cdc.Marshal(m)
 	case legacyproto.Message:
```

Comment:
do we ignore these `caseOrder: case legacyproto.Message must go before the gogoproto.Message case (gocritic)` github actions checks?

Reply:
I don't think this is needed actually, they are both the same now.

Reply:
alright then